### ISSUE TITLE
Transition to any state?

### ISSUE THREAD

**Issue by axelson** [subvisual/fsmx#2]
Is there a way to encode a transition from a state to any state, or do you have to list all the states you could transition from individually?

And is there any documentation of what the `transitions` key means exactly? I only see examples in the readme, such as:

```
  use Fsmx.Struct, transitions: %{
    "one" => ["two", "three"],
    "two" => ["three", "four"],
    "three" => "four"
  }
```

**Comment by naps62** [comment]
@axelson I thought I had replied to this yesterday, but I guess I forgot to hit the submit button. My bad

I think adding that feature makes total sense. I'm just trying to figure out how to do it

I'm hesitant to allow something like `"one" => :any`, because even though all examples use strings for state names, atoms are also perfectly valid (as long as you're not using the Ecto layer).
So this would be both a breaking change, and a confusing edge case where you could no longer have a state called `:any`

A possible approach is to just make that default behaviour when the origin state is not specified in the transitions map.

Thoughts?

**Comment by axelson** [comment]
Hmm, it would be nice to fully segment the possible "any" from any valid value, but perhaps `:*` could be used as the "any" value, I think that's significantly unlikely enough for a user to want to use as a state.

(and two days is well within a reasonable time to respond to a github issue IMO)

**Comment by naps62** [comment]
true. by saying "my bad" I meant that this almost went unnoticed, since I thought I had replied already :)

yes, `:*` sounds great. I forgot that was possible

I'll get that done tomorrow, most likely

**Comment by naps62** [comment]
@axelson added in https://github.com/subvisual/fsmx/commit/42660d2cbf085dd7c5ec85aacd54b4cc358d2c92

Also published `0.2.1`. Feel free to try it out :)

**Comment by axelson** [comment]
Looks great, thanks!
